scripts/behavior.js

```diff
@@ -371,7 +371,129 @@ function executeCommand(command, arguments, cooldown, message, settings, utils,
                 }).then(
                     callback(COMMAND_SUCCESS)
                 );
-                
+
+                break;
+
+            case "createrole":
+                if (arguments == null){
+                    callback(COMMAND_MISUSE);
+                    break;
+                }
+                let roleName = escapeArguments(arguments);
+
+                let specificsCreateRole = utils.getSpecifics(message.guild);
+
+                if(specificsCreateRole["registeredRoles"].some(role => role === roleName)) {
+                    replyToMessage(message, "Role already registered.")
+                        .then(callback(COMMAND_MISUSE));
+                    break;
+                }
+
+                specificsCreateRole["registeredRoles"].push(roleName);
+                utils.writeSpecifics(message.guild, specificsCreateRole);
+
+                message.guild.createRole({name: arguments, color: "LIGHT_GREY", mentionable: true, hoist: false}, "Created by Dostya as requested by" + message.author.username)
+                    .then(value => {
+                        replyToMessage(message, "Role created")
+                            .then(callback(COMMAND_SUCCESS));
+                    }, rejectReason => {
+                        let specificsCreateRoleFailed = utils.getSpecifics(message.guild);
+                        delete specificsCreateRoleFailed.registeredRoles[specificsCreateRoleFailed.registeredRoles.indexOf(roleName)];
+                        utils.writeSpecifics(message.guild, specificsCreateRoleFailed);
+                        replyToMessage(message, "Could not create role due to internal discord error. " + rejectReason).then(callback(COMMAND_FORBIDDEN));

```

Comment:
this one is dubious
1) in case of failure theres no need to answer at all, the bot mostly should only speak when it has something else to say than "i failed"
2) Command forbidden ? Why not COMMAND_MISUSE ? command forbidden is when the user has insufficient rights
3) "Could not create role du to internal discord error" is not good enough. The bot could be unable to create the role because of insufficient permissions, because of discord error, because of other reasons maybe - the error should help the user solve the problem. "This did not work : Maybe dostya has not enough rights ?" etc...
4) utils.log more please in that case. If creating a role is a rare thing only for moderators, since it actually touches the specifics and will leave a trace I think you can put a utils.log("This user created role <role_name>.<role_id>", "--", "message.guild")

Reply:
2. This is no misuse, this is an error (the user did nothing wrong), there is no callback response for that, the only thing fitting would be SUCCESS, but do you really wanna success in that case?
3. it is an internal call of the discord API that fails, the error is included in the message, this won't be used by users, so no need to be overfriendly
4. oops, yep, forgot logging

Reply:
be friendly with mods!